src/praisonai-agents/praisonaiagents/tools/tool_search.py

```
@@ -0,0 +1,578 @@
+            
+        Returns:
+            List of matching tool items, sorted by relevance
+        """
+        query_tokens = self._tokenize(query)
+        if not query_tokens:
+            return []
+        
+        scores = []
+        k1, b = 1.5, 0.75  # BM25 parameters
+        
+        for i, (item, tf) in enumerate(zip(self.catalog, self.term_frequencies)):
```

Ruff (0.15.14)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)